core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -0,0 +1,142 @@
+package org.broadinstitute.dsde.rawls.spendreporting
+
+import java.util
+import java.util.Currency
+
+import akka.http.scaladsl.model.StatusCodes
+import com.google.api.services.bigquery.model._
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
+import org.broadinstitute.dsde.rawls.dataaccess.{SamDAO, SlickDataSource}
+import org.broadinstitute.dsde.rawls.model._
+import org.broadinstitute.dsde.workbench.google.GoogleBigQueryDAO
+import org.broadinstitute.dsde.workbench.model.google.{BigQueryDatasetName, BigQueryTableName, GoogleProject}
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters._
+import scala.math.BigDecimal.RoundingMode
+
+object SpendReportingService {
+  def constructor(dataSource: SlickDataSource, bigQueryDAO: GoogleBigQueryDAO, samDAO: SamDAO, defaultTableName: String, serviceProject: GoogleProject)
+                 (userInfo: UserInfo)
+                 (implicit executionContext: ExecutionContext): SpendReportingService = {
+    new SpendReportingService(userInfo, dataSource, bigQueryDAO, samDAO, defaultTableName, serviceProject)
+  }
+}
+
+
+class SpendReportingService(userInfo: UserInfo, dataSource: SlickDataSource, bigQueryDAO: GoogleBigQueryDAO, samDAO: SamDAO, defaultTableName: String, serviceProject: GoogleProject)
+                           (implicit val executionContext: ExecutionContext) extends LazyLogging {
+
+  // todo: extract this out of userservice so we don't have to duplicate it?
+  def requireProjectAction[T](projectName: RawlsBillingProjectName, action: SamResourceAction)(op: => Future[T]): Future[T] = {
+    samDAO.userHasAction(SamResourceTypeNames.billingProject, projectName.value, action, userInfo).flatMap {
+      case true => op
+      case false => Future.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, "You must be a project owner.")))
+    }
+  }
+
+  def extractSpendReportingResults(rows: List[TableRow], startTime: DateTime, endTime: DateTime): SpendReportingResults = {
+    val currency = Currency.getInstance(rows.head.getF.get(2).getV.toString)
```

Comment:
🤔 How heavy of a lift would it be to check with sam to ensure the calling user is in the alpha group? I'd like to limit this API's exposure if possible

Reply:
not heavy at all, I can add that in